processing/src/main/java/org/apache/druid/segment/DimensionDictionary.java

```diff
@@ -114,6 +123,11 @@ public int add(@Nullable T originalValue)
       final int index = idToValue.size();
       valueToId.put(originalValue, index);
       idToValue.add(originalValue);
+
+      // Add size of the String and two of its references to the total size
+      // TODO: map entry overhead?
+      sizeInBytes.addAndGet(getObjectSize(originalValue) + Long.BYTES + Long.BYTES);
```

Comment:
Instead of relying on the comment to be able to read what this code is doing, you can write it as

```
long sizeOfReference = Long.BYTES;
long sizeOfString = getObjectSize(originalValue);
sizeInBytes.addAndGet(sizeOfString + (2 * sizeOfReference));
```

Now, the actual code is saying the same thing that your comment was.